src/cdk/bidi/directionality.ts

```diff
@@ -19,12 +19,32 @@ export type Direction = 'ltr' | 'rtl';
  */
 @Injectable({providedIn: 'root'})
 export class Directionality implements OnDestroy {
-  /** The current 'ltr' or 'rtl' value. */
-  readonly value: Direction = 'ltr';
+  protected _isInitialized: boolean;
+
+  /** Current layout direction of the element. */
+  get value(): Direction { return this.dir; }
 
   /** Stream that emits whenever the 'ltr' / 'rtl' state changes. */
   readonly change = new EventEmitter<Direction>();
 
+  /** @docs-private */
+  get dir(): Direction { return this._dir; }
+  set dir(value: Direction) {
```

Comment:
I don't want to expose a public setter for this since it doesn't actually set the global `dir` attribute. Additionally, this will emit when the property changes, but bot when the `dir` attribute actually changes; we would either need to set up a mutation observer for that (which I would hesitate to do for perf reasons) or say that you can _only_ update the global dir via the service

Reply:
That's why I've made `docs-private` for now. I'm fine with documenting that the changes will only be picked up if they're done programmatically through the service/directive. Right now they'll be picked up from the directive, but not the service.